pkg/skaffold/build/jib/jib_sync.go

```diff
@@ -0,0 +1,213 @@
+					if ff, err := filepath.Abs(f); err != nil{
+						return nil, nil, err
+					} else {
+						f = ff
+					}
+				}
+				if se.Src == f {
+					matches[se.Src] = []string{se.Dest}
+					break
+				}
+			}
+		}
+		if len(matches) == len(e.Modified) {
+			return matches, nil, nil
+		}
+	}
+
+	if len(e.Deleted) != 0 {
+		// change into logging
+		fmt.Println("Deletions are not supported by jib auto sync at the moment")
+		return nil, nil, nil;
+	}
+
+	// we need to do another build and get a new sync map
+	newSyncMap, err := getSyncMap(ctx, workspace, artifact)
+	if err != nil {
+		return nil, nil, err;
+	}
+	syncLists[artifact.Project] = *newSyncMap
+
+	toCopy := make(map[string][]string)
+	// calculate the diff of the syncmaps
+	// known: this doesn't handle the case that something in the oldSyncMap is
+	// no longer represented in the new sync map
+	for _, se := range newSyncMap.Generated {
+		for _, seOld := range oldSyncMap.Generated {
+			if se.Src == seOld.Src && !se.filetime.Equal(seOld.filetime) {
+				toCopy[se.Src] = []string{se.Dest}
+			}
+		}
+	}
+	for _, se := range newSyncMap.Direct {
+		for _, seOld := range oldSyncMap.Direct {
+			if se.Src == seOld.Src && !se.filetime.Equal(seOld.filetime) {
+				toCopy[se.Src] = []string{se.Dest}
+			}
+		}
+	}
+
+	return toCopy, nil, nil
+}
+
+// getSyncMap returns a list of files that can be sync'd to a remote container
+func getSyncMap(ctx context.Context, workspace string, artifact *latest.JibArtifact) (*SyncMap, error) {
+
+	// cmd will hold context that identifies the project
+	cmd, err := getSyncMapCommand(ctx, workspace, artifact)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	projectSyncMap := SyncMap{}
+	if err = runAndParseSyncMap(cmd, &projectSyncMap); err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+
+	// store the filetimes for all these values
+	if err := updateModTime(projectSyncMap.Direct); err != nil {
+		return nil, errors.WithStack(err)
+	}
+	if err := updateModTime(projectSyncMap.Generated); err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	return &projectSyncMap, nil
+}
+
+func getSyncMapCommand(ctx context.Context, workspace string, artifact *latest.JibArtifact) (*exec.Cmd, error) {
+	t, err := DeterminePluginType(workspace, artifact)
+	if err != nil {
+		return nil, err
+	}
+
+	switch t {
+	case JibMaven:
+		return getSyncMapCommandMaven(ctx, workspace, artifact), nil
+	case JibGradle:
+		return nil, errors.Errorf("unable to sync gradle projects at %s", workspace)
+	default:
+		return nil, errors.Errorf("unable to determine Jib builder type for %s", workspace)
+	}
+}
+
+func runAndParseSyncMap(cmd *exec.Cmd, sm *SyncMap) error {
+	stdout, err := util.RunCmdOut(cmd)
+	if err != nil {
+		return errors.Wrap(err, "failed to get Jib sync map")
+	}
+
+	// To parse the output, search for "BEGIN JIB JSON", then unmarshal the next line into the pathMap struct.
+	matches := regexp.MustCompile(`BEGIN JIB JSON\r?\n({.*})`).FindSubmatch(stdout)
```

Comment:
I liked @briandealwis's [idea](https://github.com/GoogleContainerTools/jib/pull/2192#issuecomment-563303984) to have this extensible. Something we should keep in mind to make things match.